### PR TITLE
fix: Update CI to use kokoro and build for OSX, Windows, Linux

### LINKED ISSUE #149: Migrate Release Tooling
Creating an issue to track migrating the release tooling from [GH Action](https://github.com/googleapis/python-crc32c/blob/main/.github/workflows/python-publish.yml) to Kokoro.

### PR DESCRIPTION
**Pull request by vishwarajanand**
Changes:

1. This PR shows the placeholder changes intended for migrating GH actions build to an internal release platform called kokoro.
2. Maybe fixes: https://github.com/googleapis/python-crc32c/issues/149
3. We plan to do a `dev` package release with this branch

Note: 
1. Some cleanup might be needed in this PR and this is not the .

### CODE REVIEW

File: .kokoro/build.sh [hunk 1]
```diff
@@ -57,3 +68,19 @@ if [[ -n "${NOX_SESSION:-}" ]]; then
...
+# python3.6 -m pip install --quiet git+https://github.com/theacodes/ci-diff-helper.git
+# python3.6 test_utils/scripts/get_target_packages_kokoro.py > ~/target_packages
+# cat ~/target_packages
+
+# if [[ ! -n $(grep -x "$PACKAGE" ~/target_packages) ]]; then
+#     echo "$PACKAGE was not modified, returning."
+#     exit;
+# fi
+
+# cd "$PACKAGE"
```
Comment (parthea) [inline .kokoro/build.sh:82]: Delete instead of commenting out

File: .kokoro/build.sh [hunk 2]
```diff
@@ -57,3 +68,19 @@ if [[ -n "${NOX_SESSION:-}" ]]; then
...
+
+# Some system tests require indexes. Use gcloud to create them.
+gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS --project=$PROJECT_ID
+#gcloud --quiet --verbosity=debug datastore indexes create tests/system/index.yaml
```
Comment (parthea) [inline .kokoro/build.sh:86]: Delete instead of commenting out

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""py.test shared testing configuration.
+
+This
+
+* Monkey-patches ``mock`` as ``unittest.mock`` for Python 2.7.
```
Comment (parthea) [inline tests/conftest.py:19]: Is this needed?

File: src/google_crc32c/__config__.py [hunk 4]
```diff
@@ -35,12 +34,13 @@ def modify_path():
             from importlib_resources import files as _resources_files  # type: ignore
         extra_dll_dir = str(_resources_files("google_crc32c") / "extra-dll")
         if os.path.isdir(extra_dll_dir):
-            # Python 3.7 use path
+            # Python 3.6, 3.7 use path
```
Comment (parthea) [inline src/google_crc32c/__config__.py:37]: revert

File: scripts/windows/build.bat [hunk 5]
```diff
@@ -15,28 +15,26 @@
...
-   set PYTHON_VERSION=3.10
+  echo "Python version was not provided, using Python 3.10"
+  set PYTHON_VERSION=3.10
 )
 
 py -0
 py -%PYTHON_VERSION% -m pip install cmake
 
+@rem Add directory as safe to avoid "detected dubious ownership" fatal issue
+@REM git config --global --add safe.directory C:/tmpfs/src/github/python-crc32c
```
Comment (parthea) [inline scripts/windows/build.bat:36]:
```suggestion
```

File: tests/conftest.py (hunk 3 above)
Comment (vishwarajanand) [reply tests/conftest.py]: removed.. it was causing test failure

File: scripts/windows/build.bat (hunk 5 above)
Comment (vishwarajanand) [reply scripts/windows/build.bat]: This was required.. so moved back from `@REM` to being called now.

File: scripts/windows/build.bat [hunk 6]
```diff
@@ -15,40 +15,35 @@
...
 set CONFIGURATION=RelWithDebInfo
 set CRC32C_INSTALL_PREFIX=%cd%\build\%CONFIGURATION%
 
-@rem Path to cmake, env var to make it easier to point to a specific version
-set cmake=cmake
+@rem Iterate through supported Python versions.
+@rem Unfortunately pyenv for Windows has an out-of-date versions list. Choco's
+@rem installer seems to have some problems with installing multiple versions at
+@rem once, so as a workaround, we will install and then uninstall every version.
+FOR %%P IN (3.8, 3.9, 3.10, 3.11, 3.12) DO (
```
Comment (edgarrmondragon) [inline scripts/windows/build.bat:26]: chocolatey seems to translate, for example, `3.8` to `3.8.0` (see https://github.com/chocolatey/choco/issues/800). So you might wanna use more specific versions here:

```suggestion
FOR %%P IN (3.8.19, 3.9.19, 3.10.14, 3.11.9, 3.12.4) DO (
```

Comment (andrewsg) [reply scripts/windows/build.bat:26]: Yes, unfortunately, making that change will break the rest of the code around python versions, so I will need to do a translation. It's on the docket for later.

File: scripts/osx/build_python_wheel.sh [hunk 7]
```diff
@@ -50,14 +69,23 @@ ${VENV}/bin/python setup.py build_ext \
...
     ${DIST_WHEELS}/google_crc32c*${PY_TAG}*.whl
 
+# Upload wheels to GCS for debugging. Uncomment only when needed.
+
+export GOOGLE_APPLICATION_CREDENTIALS=${KOKORO_GFILE_DIR}/service-account.json
+gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+ls ${REPO_ROOT}/wheels/
+gsutil cp ${REPO_ROOT}/wheels/* gs://python_crc32c/
+echo "Osx wheels uploaded successfully"
+
```
Comment (parthea) [inline scripts/osx/build_python_wheel.sh:88]:
```suggestion
```

Remove test code

**Comment by parthea** [comment]
Superseded by https://github.com/googleapis/python-crc32c/pull/234

**Pull request closed without merging**